cmd/kubeadm/app/phases/upgrade/health.go

```diff
@@ -101,18 +101,19 @@ func createJob(client clientset.Interface, cfg *kubeadmapi.ClusterConfiguration)
 	// If client.Discovery().RESTClient() is nil, the fake client is used.
 	// Return early because the kubeadm dryrun dynamic client only handles the core/v1 GroupVersion.
 	if client.Discovery().RESTClient() == nil {
-		fmt.Printf("[upgrade/health] Would create the Job %q in namespace %q and wait until it completes\n", jobName, ns)
+		fmt.Printf("[upgrade/health] Would create the Job with prefix %q in namespace %q and wait until it completes\n", jobName, ns)
 		return nil
 	}
 
 	// Prepare Job
 	job := &batchv1.Job{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      jobName,
-			Namespace: ns,
+			GenerateName: jobName + "-",
```

Comment:
can we preserve the job name the way it is today?

Reply:
@neolit123 

if reuse the job name, we may have to delete the existing job when we recreate it.  i.e. the existing job was created but maybe not removed by kubeadm.

Reply:
ok, i see.